### ISSUE TITLE
[FUNCTIONALITY] Sortable Table - SHOULD provide a sorting function

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#9194]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️  Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Our [sortable table component](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design/components/sortabletable/) has clickable `<th>` elements but does not provide a sorting function. If we are going to provide this component, we need to include a base sort, and allow developers to pass in their own overrides as needed.

## Related Issues
* #9193 

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [ ] Basic sorting functionality is added to component
- [ ] Sorting function can be overridden by passing in a `prop`
- [ ] Unit testing is added to confirm sort ascending and descending behavior
- [ ] Sorting behavior should be defined for String, Number, and maybe Boolean

## Environment

* https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design/components/sortabletable/

## WCAG or Vendor Guidance (optional)

* [Smashing Magazine: Creating Sortable Tables With React](https://www.smashingmagazine.com/2020/03/sortable-tables-react/)

**Comment by 1Copenut** [comment]
Issue moved to [department-of-veterans-affairs/vets-design-system-documentation #230](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/230) via [**ZenHub**](https://www.zenhub.com/)
